astropy/wcs/wcs.py

```diff
@@ -389,6 +389,9 @@ def __init__(self, header=None, fobj=None, key=' ', minerr=0.0,
                     header_string = header
             elif isinstance(header, fits.Header):
                 header_string = header.tostring().rstrip()
+            elif type(header) == fits.ImageHDU or type(header) == fits.PrimaryHDU:
+                header = header.header
+                header_string = header.tostring().rstrip()
```

Comment:
But in first clause of this conditional when ``is_path`` is ``False`` than ``header_string = header`` which cant be resolved by moving ``header_string = header.tostring().rstrip()`` to last.
How to resolve that?